enumerables.rb

```diff
@@ -0,0 +1,177 @@
+    if !argument.nil?
+      if argument.is_a?(Regexp)
+        arr.my_each { |current_element| all_true = argument.match(current_element) ? true : all_true }
+      elsif argument.class == Class
+        arr.my_each { |current_element| all_true = current_element.is_a?(argument) ? true : all_true }
+      else
+        arr.my_each { |current_element| all_true = current_element == argument ? true : all_true }
+      end
+    elsif block_given? && argument.nil?
+      arr.my_each { |condition| all_true = yield(condition) || all_true }
+      all_true
+    elsif !block_given? && argument.nil?
+      arr.my_each { |current_element| all_true = current_element ? true : all_true }
+    end
+    all_true
+  end
+
+  def my_none?(argument = nil)
+    arr = self
+    arr = arr.is_a?(Array) ? arr : arr.to_a
+    all_true = true
+    if !argument.nil?
+      if argument.is_a?(Regexp)
+        arr.my_each { |current_element| all_true = argument.match(current_element) ? false : all_true }
+      elsif argument.class == Class
+        arr.my_each { |current_element| all_true = current_element.is_a?(argument) ? false : all_true }
+      else
+        arr.my_each { |current_element| all_true = current_element == argument ? false : all_true }
+      end
+    elsif block_given? && argument.nil?
+      arr.my_each { |condition| all_true = yield(condition) ? false : all_true }
+    elsif !block_given? && argument.nil?
+      arr.my_each { |current_element| all_true = current_element ? false : all_true }
+    end
+    all_true
+  end
+
+  def my_count(argument = nil)
+    counter = 0
+    arr = self
+    if argument.nil?
+      if block_given?
+        arr.my_each { |current_element| counter += 1 if yield(current_element) }
+      else
+        arr.my_each { counter += 1 }
+      end
+    else
+      arr.my_each { |current_element| counter += 1 if current_element == argument }
+    end
+    counter
+  end
+
+  def my_inject(num = nil, symb = nil)
+    arr = self
+    arr = arr.is_a?(Array) ? arr : arr.to_a
+    if num.nil? and symb.nil?
+      accumulator = arr[0]
+      if block_given?
+        (arr.length - 1).times do |i|
+          accumulator = yield(accumulator, arr[i + 1])
+        end
+      end
+      accumulator
+    elsif num and symb
+      unless block_given?
+        accumulator = num
+        arr.length.times { |i| accumulator = accumulator.send(symb, arr[i]) }
+        accumulator
+      end
+    elsif num.nil? or symb.nil?
+      if num.is_a?(Symbol)
+        unless block_given?
+          accumulator = arr[0]
+          symbol = num
+          (arr.length - 1).times do |i|
+            accumulator = accumulator.send(symbol, arr[i + 1])
+          end
+          accumulator
+        end
+      elsif num.is_a?(Integer)
+        if block_given?
+          accumulator = num
+          arr.length.times do |i|
+            accumulator = yield(accumulator, arr[i])
+          end
+          accumulator
+        end
+      end
+    end
+  end
```

Comment:
# my_inject 
1. [ ] Raises a "LocalJumpError" when no block or argument is given
### Test 👉 ` expect { array.my_inject }.to raise_error(LocalJumpError)`
#### Expected `LocalJumpError` but nothing was raised.

Reply:
Implemented this